3.6/aql/execution-and-performance-optimizer.md

```diff
@@ -322,145 +335,142 @@ List of execution nodes
-* *GatherNode*: used on a coordinator to aggregate results from one or many shards
+- *ScatterNode*: used on a coordinator to fan-out data to one or multiple shards.
+- *GatherNode*: used on a coordinator to aggregate results from one or many shards
   into a combined stream of results.
-* *DistributeNode*: used on a coordinator to fan-out data to one or multiple shards,
+- *DistributeNode*: used on a coordinator to fan-out data to one or multiple shards,
   taking into account a collection's shard key.
-* *RemoteNode*: a *RemoteNode* will perform communication with another ArangoDB
+- *RemoteNode*: a *RemoteNode* will perform communication with another ArangoDB
   instances in the cluster. For example, the cluster coordinator will need to
   communicate with other servers to fetch the actual data from the shards. It
   will do so via *RemoteNode*s. The data servers themselves might again pull
   further data from the coordinator, and thus might also employ *RemoteNode*s.
   So, all of the above cluster relevant nodes will be accompanied by a *RemoteNode*.
 
-
 List of optimizer rules
 -----------------------
 
 The following optimizer rules may appear in the `rules` attribute of a plan:
 
-* `move-calculations-up`: will appear if a *CalculationNode* was moved up in a plan.
+- `fuse-filters`: will appear if the optimizer merges adjacent FILTER nodes together into
+   a single FILTER node
+- `geo-index-optimizer`: will appear when a geo index is utilized.
+- `inline-subqueries`: will appear when a subquery was pulled out in its surrounding scope,
+  e.g. `FOR x IN (FOR y IN collection FILTER y.value >= 5 RETURN y.test) RETURN x.a`
+  would become `FOR tmp IN collection FILTER tmp.value >= 5 LET x = tmp.test RETURN x.a`
+- `interchange-adjacent-enumerations`: will appear if a query contains multiple
+  *FOR* statements whose order were permuted. Permutation of *FOR* statements is
+  performed because it may enable further optimizations by other rules.
+- `move-calculations-down`: will appear if a *CalculationNode* was moved down in a plan.
+  The intention of this rule is to move calculations down in the processing pipeline
+  as far as possible (below *FILTER*, *LIMIT* and *SUBQUERY* nodes) so they are executed
+  as late as possible and not before their results are required.
+- `move-calculations-up`: will appear if a *CalculationNode* was moved up in a plan.
   The intention of this rule is to move calculations up in the processing pipeline
   as far as possible (ideally out of enumerations) so they are not executed in loops
   if not required. It is also quite common that this rule enables further optimizations
   to kick in.
-* `move-filters-up`: will appear if a *FilterNode* was moved up in a plan. The
+- `move-filters-into-enumerate`: moves filters on non-indexed collection
+  attributes into *IndexNode* or *EnumerateCollectionNode* to allow early pruning
+  of non-matching documents.
```

Comment:
https://github.com/arangodb/arangodb/pull/10337/files
```suggestion
  of non-matching documents. This optimization can help to avoid a lot of
  temporary document copies.
```